src/test/javascript/spec/component/grading-system/grading-key-overview.component.spec.ts

```diff
@@ -47,7 +48,21 @@ describe('GradeKeyOverviewComponent', () => {
         maxPoints: 100,
     };
 
+    const studentGrade = '2.0';
+
     beforeEach(() => {
+        route = {
+            snapshot: { params: {}, queryParams: { grade: studentGrade } },
+            parent: {
+                snapshot: { params: {} },
+                parent: {
+                    snapshot: {
+                        params: { courseId: 345, examId: 123 },
+                    },
+                },
+            },
+        } as unknown as ActivatedRoute;
```

Comment:
Is the `as unknown` part needed?
```suggestion
        } as ActivatedRoute;
```

Reply:
Only making this change causes linter to complain, I also added `as Params` type casts above.